src/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -0,0 +1,86 @@
+        /// <param name="createdNew">When this method returns, it is set to <see langword="true" /> if a local event was created (that is, if name is <see langword="null" /> or an empty string) or if the specified named system event was created; <see langword="false" /> if the specified named system event already existed. This parameter is passed uninitialized.</param>
+        /// <param name="eventSecurity">The Windows access control security to apply.</param>
+        /// <returns>An object that represents the named event wait handle.</returns>
+        /// <exception cref="ArgumentException">The length of the name exceeds the maximum limit.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="eventSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The <paramref name="mode" /> enum value was out of legal range.</exception>
+        /// <exception cref="DirectoryNotFoundException">Could not find a part of the path specified in <paramref name="name" />.</exception>
+        /// <exception cref="WaitHandleCannotBeOpenedException">A system-wide synchronization event with the provided <paramref name="name" /> was not found.
+        /// -or-
+        /// An <see cref="EventWaitHandle" /> with system-wide name <paramref name="name" /> cannot be created. An <see cref="EventWaitHandle" /> of a different type might have the same name.</exception>
+        public static unsafe EventWaitHandle Create(bool initialState, EventResetMode mode, string name, out bool createdNew, EventWaitHandleSecurity eventSecurity)
+        {
+            if (name.Length > Interop.Kernel32.MAX_PATH)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, name), nameof(name));
+            }
+            if (eventSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(eventSecurity));
+            }
+
+            bool isManualReset = mode switch
+            {
+                EventResetMode.ManualReset => true,
+                EventResetMode.AutoReset => false,
+                _ => throw new ArgumentOutOfRangeException(nameof(mode), SR.Format(SR.ArgumentOutOfRange_Enum))
+            };
+
+            fixed (byte* pSecurityDescriptor = eventSecurity.GetSecurityDescriptorBinaryForm())
+            {
+                var secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
+                {
+                    nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
+                    lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
+                };
+
+                SafeWaitHandle handle = Interop.Kernel32.CreateEvent(ref secAttrs, isManualReset, initialState, name);
+
+                ValidateHandle(handle, name, out createdNew);
+
+                var ewh = new EventWaitHandle(initialState, mode);
```

Comment:
I don't like creating global objects when they aren't needed. Can we condition this on their being a name?

Additionally, as I stated, I'd rather modify the existing security when there is a name, for the same reason- I think it's better not to create throwaway global objects if possible.

Reply:
@JeremyKuhne, can you help me understand better why?  It's unnamed and very short-lived.  It's also faster:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Diagnosers;
using BenchmarkDotNet.Running;
using System;
using System.Threading;

[MemoryDiagnoser]
public class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssemblies(new[] { typeof(Program).Assembly }).Run(args);

    private static readonly string s_name = Guid.NewGuid().ToString("N");
    private EventWaitHandle _named = new EventWaitHandle(false, EventResetMode.ManualReset, s_name);

    [Benchmark]
    public void CreateNew() => new EventWaitHandle(false, EventResetMode.ManualReset).Dispose();

    [Benchmark]
    public void OpenExisting() => EventWaitHandle.OpenExisting(s_name).Dispose();
}
```

|       Method |     Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------- |---------:|----------:|----------:|-------:|------:|------:|----------:|
|    CreateNew | 1.346 us | 0.0041 us | 0.0034 us | 0.0076 |     - |     - |      56 B |
| OpenExisting | 2.121 us | 0.0069 us | 0.0057 us | 0.0076 |     - |     - |      56 B |

Reply:
My primary concern is that we're forcing more contention in creating twice as many synchronizing objects. To be fair, I haven't spent much time looking at the Windows code here, so it is more of a gut concern. If you feel comfortable with it, I'm ok with trying the current route.